### ISSUE TITLE
ML-Benchmark prediction submission: [UiB-CorrDiff]

### ISSUE THREAD

**Issue by joshdorrington** [WCRP-CORDEX/ml-benchmark#17]
### Institution ID

UiB

### Institution Name

University of Bergen

### Emulator identifier (`emulator_id`)

UiBCorrDiff-v1-orog

### Emulator Description

 Corrective Diffusion model using a regression U-Net and a residual EDM diffusion U-Net. The regression model is trained with Tweedie deviance loss with scale parameter p=1.6 for pr and with MSE loss for tasmax. The diffusion model is trained using a score matching loss for both variables. Both modules use a 128 channel, 5-block U-Net architecture with 8 group norm + SiLu + 3-point convolution steps in each block of the descending and ascending branches.

All CordexBench predictors + static orography were used to predict both tasmax and pr. Predictor data and tasmax predictand data were standardised to mean=0 variance=1 relative to the training data. pr predictand data was scaled by dividing by global 99th percentile of gridpoint pr in the training data. The regression model was trained for 128 000 data samples with lr=2e-4 and batch size 64. The diffusion model was trained for 1 152 000 data samples with lr=2e-3 and batch size 256.

The ensemble was generated using different random seeds passed to a deterministic sampler. The sampler used 36 denoising steps, and a scale parameter rho=9.

### Hardware and Training Details

Hardware: One NVIDIA GH200 120GiB GPU. Training time: 33 min/128,000 samples for Regression module and 25 min/128,000 samples for Diffusion module. Inference time: 3.8 min/year of predictions.

### Stochastic/Probabilistic Output

yes

### Reference URL

https://doi.org/10.1038/s43247-025-02042-5

### Repository for Reproducibility

https://github.com/joshdorrington/UiBcorrdiff

### Additional Notes

In the process of adding our code to the UiBcorrdiff repository currently.

**Comment by joshdorrington** [comment]
I had to move the data to long-term storage after submission, so please add the new pubkey I submitted with the form, and I will send the zip file across again.

**Comment by jgonzalezab** [comment]
We have just added the key, so you can go ahead and submit it.
